### PR TITLE
fix: string to number erange

### PR DESCRIPTION
**Pull request by jun0315**
see https://github.com/4paradigm/OpenMLDB/issues/2252

### CODE REVIEW

**Comment by vagetablechicken** [comment]
merge the main, if still has failed tests in ci, plz check the test @aceforeverd

**Comment by dl239** [comment]
please add the test case

**Comment by aceforeverd** [comment]
> ## HybridSE Linux Test Report
> 19 437 tests   19 429 [✔️](https://github.com/EnricoMi/publish-unit-test-result-action/blob/v1.20/README.md#the-symbols)  7m 24s [⏱️](https://github.com/EnricoMi/publish-unit-test-result-action/blob/v1.20/README.md#the-symbols)      244 suites           2 [💤](https://github.com/EnricoMi/publish-unit-test-result-action/blob/v1.20/README.md#the-symbols)        67 files             6 [❌](https://github.com/EnricoMi/publish-unit-test-result-action/blob/v1.20/README.md#the-symbols)
> 
> For more details on these failures, see [this check](https://github.com/4paradigm/OpenMLDB/runs/12077023379).
> 
> Results for commit [e79cd2b](https://github.com/4paradigm/OpenMLDB/commit/e79cd2bb2e267cefcb598683997b9a4c559c5f56).
> 
> ♻️ This comment has been updated with latest results.

You can see all failed tests: https://github.com/4paradigm/OpenMLDB/runs/12077023379

```
[==========] 420 tests from 1 test suite ran. (5871 ms total)
[  PASSED  ] 416 tests.
[  FAILED  ] 4 tests, listed below:
[  FAILED  ] EngineTestCalculateFunction/EngineTest.TestBatchEngine/3, where GetParam() = Case ID: 3, Desc:反三角函数-三角函数-常量
[  FAILED  ] EngineTestCalculateFunction/EngineTest.TestBatchRequestEngineForLastRow/3, where GetParam() = Case ID: 3, Desc:反三角函数-三角函数-常量
[  FAILED  ] EngineTestCalculateFunction/EngineTest.TestClusterRequestEngine/3, where GetParam() = Case ID: 3, Desc:反三角函数-三角函数-常量
[  FAILED  ] EngineTestCalculateFunction/EngineTest.TestClusterBatchRequestEngine/3, where GetParam() = Case ID: 3, Desc:反三角函数-三角函数-常量
```

File: hybridse/src/udf/udf.cc [hunk 1]
```diff
@@ -653,33 +654,24 @@ void string_to_int(StringRef *str, int32_t *out, bool *is_null_ptr) {
...
+    std::string str_obj = str->ToString();
+    const char *c_str = str_obj.c_str();
+    char *end;
+    *out = strtol(c_str, &end, 10);
+    if (end < c_str + str->size_) {
+        *out = 0;
+        *is_null_ptr = true;
+        return;
+    }
+    if (errno == ERANGE) {
```
Comment (aceforeverd) [inline hybridse/src/udf/udf.cc:666]: detect error for `strtoXX` is complex, you can see 
- https://stackoverflow.com/a/26083517/6187385

To be simple, try `absl::SimpleAtoi`

**Pull request closed without merging**